cloudcompose/datadog/monitoring/datadogcontroller.py

```diff
@@ -1,135 +1,139 @@
+                'monitor:{}'.format(monitor_tag)
+            ]   
+            match = self._get_existing_monitor(tags)
+
+            if match:
+                if 'id' in match:
+                    monitor_id = match['id']
+                    print 'Deleting monitor {}'.format(monitor_tag)
+                    api.Monitor.delete(monitor_id)
+
+    def _resolve_monitor(self, monitor):
+        # Take the monitor definition from the yaml and transform it 
+        #       to what will be used to create the actual monitor
+        monitor['tags'] = ['clustername:{}'.format(self.cluster_name),
+                'monitor:{}'.format(monitor.get('tag'))]
+        monitor['query'] = monitor.get('query') % self.config_data
+        cluster_prefix = self.config_data.get('use_cluster_prefix', True)
+        if cluster_prefix:
+            monitor['name'] = '[{}] {}'.format(self.cluster_name, monitor.get('name'))
+
+        notified = self.config_data.get('notify', [''])
+        monitor['message'] = '{} {}'.format(monitor.get('message'), ' '.join(notified))
+
+        # monitor['options'] = {}
+        monitor['options'] = monitor.get('options', {}) # Instantiate it to an empty dict if it's not present
+        for option in self.default_options:
+            default_val = self.default_options[option]
+            # Adds any default options not defined on the monitor level to the monitor
+            monitor['options'][option] = monitor['options'].get(option, default_val) 
+
+    def _create_monitors(self):
+        for monitor in self.monitor_data:
+            self._resolve_monitor(monitor)
+            monitor_tag = monitor.get('tag')
+            tags = monitor.get('tags')
+            message = monitor.get('message')
+            query = monitor.get('query')
+            name = monitor.get('name')
+            options = monitor.get('options')
+
+            old_monitor = self._get_existing_monitor(tags)
+
+            if old_monitor:
+                # Update rather than create
+                monitor_id = old_monitor['id']
+                print 'Updating monitor {}:'.format(monitor_tag)
+                response = api.Monitor.update(id=monitor_id, query=query, name=name, message=message, options=options, tags=tags)
+                # self.pp.pprint(monitor)
+                self.pp.pprint(response)
+            else:
+                print 'Creating monitor {} for cluster {}:'.format(monitor_tag, self.cluster_name)
+                response = api.Monitor.create(type=TYPE, query=query, name=name, message=message, options=options, tags=tags)
+                # self.pp.pprint(monitor)
+                self.pp.pprint(response)
+            print '-'*16
+
+
+    def _get_existing_monitor(self, tags):
+        tags = sorted(tags)
+        try:
+            monitors = api.Monitor.get_all(type=TYPE)
+        except:
+            raise CloudComposeException('ERROR: DATADOG_API_KEY or DATADOG_APP_KEY not valid')
```

Comment:
catch the exception and put the message into the CloudComposeException. Then add a note saying they should check that both DATADOG_API_KEY and DATADOG_APP_KEY are set to the correct values.